### PR TITLE
Create sever group

### PR DESCRIPTION
**Pull request by BrunoCarrier**
Create sever group

@dkirillov

### CODE REVIEW

**Comment by BrunoCarrier** [comment]
image

**Comment by BrunoCarrier** [comment]
This is relevant to both https://jira.corp.lookout.com/browse/CD-1154 and https://jira.corp.lookout.com/browse/CD-1139

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java [hunk 1]
```diff
@@ -16,23 +16,36 @@
...
+
+  ServerGroup.Capacity capacity;
+
+  Source source;
+
+  Map<String, List<String>> availabilityZones;
+
+  @Data
+  public class Source {
+    String asgName, serverGroupName, account, region, useSourceCapacity;
```
Comment (dkirillov) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java]: `region` is already in `AbstractECSDescription`

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java [hunk 2]
```diff
@@ -48,50 +50,88 @@ private static Task getTask() {
...
+    return result;
   }
 
-  private TaskDefinition registerTaskDefinition(AmazonECS ecs) {
+  private String inferServergroupVersion() {
+    if (description.getSource() == null) {
+      return "v0001";
+    } else {
+      String[] splitName = description.getSource().getServerGroupName().split("-");
+      String lastPortionOfSplitName = splitName[splitName.length - 1];
```
Comment (dkirillov) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java]: The above two lines can be replaced with:
```
String versionString = StringUtils.substringAfterLast(description.getSource().getServerGroupName(), "-");
```

Apache StringUtils lang3.

Optional.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java [hunk 3]
```diff
@@ -48,50 +50,88 @@ private static Task getTask() {
...
+      lastPortionOfSplitName = lastPortionOfSplitName.replace("v", "");
+
+      String numberAsText = String.valueOf(Integer.valueOf(lastPortionOfSplitName) + 1);
+      while (numberAsText.length() < 4) {
+        numberAsText = "0" + numberAsText;
+      }
+
+      String newVersion = "v" + numberAsText;
```
Comment (dkirillov) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java]: The while loop and newVersion can be replaced with:

```
String newVersion  = String.format("v%04d", (Integer.valueOf(lastPortionOfSplitName) + 1));
```

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java [hunk 4]
```diff
@@ -19,13 +19,7 @@
...
 import com.amazonaws.services.ec2.model.InstanceStatus;
 import com.amazonaws.services.ecs.AmazonECS;
-import com.amazonaws.services.ecs.model.DescribeTasksRequest;
-import com.amazonaws.services.ecs.model.DescribeTasksResult;
-import com.amazonaws.services.ecs.model.ListServicesRequest;
-import com.amazonaws.services.ecs.model.ListServicesResult;
-import com.amazonaws.services.ecs.model.ListTasksRequest;
-import com.amazonaws.services.ecs.model.ListTasksResult;
-import com.amazonaws.services.ecs.model.Task;
+import com.amazonaws.services.ecs.model.*;
```
Comment (dkirillov) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java]: Star import...

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java (hunk 1 above)
Comment (BrunoCarrier) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java]: This is different, this is about the other deployments

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java (hunk 4 above)
Comment (BrunoCarrier) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java]: haha

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java (hunk 1 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java]: Ah, alright. All good then 👍

Comment (BrunoCarrier) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/description/CreateServerGroupDescription.java]: Also, for some reason, region was not being passed anymore so i took it from the AZs.  weird, but we will have time to revisit that in teh future

**Pull request merged**